ZuluContent/Skills/AnimalTaming.cs

```diff
@@ -1,338 +1,292 @@
-						return true;
-
-					MovementPath path = new MovementPath( m_Creature, new Point3D( p ) );
-					return path.Success;
-				}
-			}
-		}
-	}
-}
+    {
+        private static readonly int MaxDistance = 20;
+        private static readonly int PrevTamedMinus = 20;
+        private static readonly TimeSpan DelayBetweenSpeech = TimeSpan.FromSeconds(3.0);
+        private static readonly int PointMultiplier = 15;
+        private static readonly TimeSpan UnresponsiveTime = TimeSpan.FromSeconds(300.0);
+
+
+        private static Hashtable m_BeingTamed = new Hashtable();
+
+        public static void Initialize()
+        {
+            SkillInfo.Table[(int) SkillName.AnimalTaming].Callback = OnUse;
+        }
+
+        public static TimeSpan OnUse(Mobile m)
+        {
+            m.RevealingAction();
+
+            m.Target = new InternalTarget();
+            m.RevealingAction();
+
+            m.SendLocalizedMessage(502789); // Tame which animal?
+
+            return Configs[SkillName.AnimalTaming].Delay;
+        }
+
+        public static bool AngerBeast(BaseCreature creature, Mobile from, bool isClassedRanger)
+        {
+            var chance = 75;
+            if (isClassedRanger)
+                chance = (int) (chance / ClasseBonus);
+
+            if (Utility.Random(100) < chance)
+            {
+                creature.PublicOverheadMessage(MessageType.Regular, 0x3B2, true,
+                    $"{from.Name} has angered the beast!");
+                creature.PlaySound(creature.GetAngerSound());
+                creature.Direction = creature.GetDirectionTo(from);
+                creature.BardEndTime = DateTime.Now;
+                creature.BardPacified = false;
+                creature.DoHarmful(from, true);
+                return true;
+            }
+
+            return false;
+        }
+
+        public static void CalmBeast(BaseCreature creature, Mobile from)
+        {
+            PacifyBeast(creature, from);
+            creature.PublicOverheadMessage(MessageType.Regular, 0x3B2, true,
+                $"{from.Name} has calmed the beast!");
+        }
+
+        public static void PacifyBeast(BaseCreature creature, Mobile from)
+        {
+            from.Combatant = null;
+            creature.Combatant = null;
+            creature.Warmode = false;
+            creature.Pacify(from, DateTime.Now + TimeSpan.FromSeconds(1.0));
+        }
+
+        private class InternalTarget : Target
+        {
+            public InternalTarget() : base(MaxDistance, false, TargetFlags.None)
+            {
+            }
+
+            protected override void OnTarget(Mobile from, object targeted)
+            {
+                from.RevealingAction();
+
+                if (!(targeted is BaseCreature))
+                {
+                    from.SendAsciiMessage(33, "You cannot tame this!");
+                    return;
+                }
+
+                BaseCreature creature = (BaseCreature) targeted;
```

Comment:
Use pattern matching!

```csharp
                if (!(targeted is BaseCreature creature))
                {
                    from.SendAsciiMessage(33, "You cannot tame this!");
                    return;
                }

// use `creature` here
```

Reply:
Mmm but I don't need to use an if statement here which is why I casted it and wanted to prevent unnecessary nesting... what is the benefit of doing it that way here?

Reply:
It's a nitpick for sure but just pointed it out so you get into the habit of using pattern matching. It wouldn't be nesting any more than it currently is, it'd just make that `is BaseCreature` type-check also produce a variable named `creature` you could use in the following code.

There's an argument that casting inside an if statement that doesn't use the variable is bit indirect though.